linux/deploy_vm/reconfigure_vm_with_cloudinit.yml

```diff
@@ -0,0 +1,168 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Set fact of the deploy VM test case name"
+  ansible.builtin.set_fact:
+    deploy_casename: "deploy_{{ ova_guest_os_type }}_ova"
+
+- name: "Set fact of cloud-init final message"
+  ansible.builtin.set_fact:
+    cloudinit_final_msg: "VM deployed from OVA is ready for use"
+
+# Add serial port for Ubuntu OVA when hardware_version > 10.
+# Otherwise, it would hang at boot time
+- name: "Add serial port for Ubuntu VM deployed from OVA"
+  include_tasks: ../../common/vm_add_serial_port.yml
+  when:
+    - ova_guest_os_type == 'ubuntu'
+    - vm_hardware_version_num is defined
+    - vm_hardware_version_num | int > 10
+
+- name: "Create cloud-init seed ISO to configure guest OS"
+  include_tasks: ../utils/create_seed_iso.yml
+  vars:
+    user_data_template: "{{ ova_guest_os_type }}-ova-user-data.j2"
+    local_hostname: "{{ ova_guest_os_type }}-ova-{{ hostname_timestamp }}"
+    mode: '0644'
+
+- name: "Upload cloud-init seed ISO to ESXi server datastore"
+  include_tasks: ../../common/esxi_upload_datastore_file.yml
+  vars:
+    src_file_path: "{{  seed_iso_path }}"
+    dest_file_path: "{{ vm_dir_name }}/{{ seed_iso_path | basename }}"
+
+- name: "Initialize CDROM device for attaching cloud-init seed ISO"
+  ansible.builtin.set_fact:
+    vm_cdrom_controller_type: "sata"
+    vm_cdrom_controller_num: 0
+    vm_cdrom_unit_num: 0
+
+- name: "Get VM's CDROM devices"
+  include_tasks: ../../common/vm_get_cdrom_devices.yml
+
+- name: "Set fact of VM's existing CDROM device"
+  ansible.builtin.set_fact:
+    vm_existing_cdrom_list: "{{ cdrom_device_list }}"
+
+# Amazon Linux and VMware Photon OS VM can use existing CDROM device
+- name: "Use existing CDROM device for attaching cloud-init seed ISO"
+  ansible.builtin.set_fact:
+    vm_cdrom_controller_type: "{{ vm_existing_cdrom_list[0].controller_label.split()[0] |
```

Comment:
It seems this 'default' value is not needed.

Reply:
Removed 'default' value.